### PR TITLE
os/se: SE 1st release for RTL8720E

### PR DESCRIPTION
**Pull request by ZhenBei-Sin**
deleted the image file /gnu_utility/km4_image3_all.bin
removed non-trustzone not maintaining files
optimized code side and performance

### CODE REVIEW

File: build/configs/rtl8720e/scripts/rlx8720e_img2_is.ld [hunk 1]
```diff
@@ -69,7 +69,7 @@ MEMORY
 	KM4_FLOADER_RAM_S (rwx) :               ORIGIN = 0x3000A020, LENGTH = 0x30020000 - 0x3000A020	/* KM4 ImgTool Flash Loader RAM: 88k */
 	KM4_BD_RAM_NSC (rwx) :                  ORIGIN = 0x6015B020, LENGTH = 0x6015C000 - 0x6015B020	/* KM4 BD RAM NSC: 4k */
 	KM4_BD_RAM_ENTRY (rwx) :                ORIGIN = 0x6015C000, LENGTH = 0x60160000 - 0x6015C000	/* KM4 BD RAM ENTRY: 16k */
-	KM4_BD_RAM_S (rwx) :                    ORIGIN = 0x70160020, LENGTH = 0x70180000 - 0x70160020	/* KM4 BD RAM S: 128k */
+	KM4_BD_RAM_S (rwx) :                    ORIGIN = 0x70160020, LENGTH = 0x7018D000 - 0x70160020	/* KM4 BD RAM S: 128k */
```
Comment (sunghan-chang) [inline build/configs/rtl8720e/scripts/rlx8720e_img2_is.ld:72]: @ZhenBei-Sin Could you update the value in the comment? It's still 128k.
@fengyunlai Could you check your peripheral change as well? PR #5560 did not update it.

File: os/se/ameba/Make.defs [hunk 2]
```diff
@@ -41,3 +44,17 @@ CFLAGS += -I$(TOPDIR)/board/rtl8721csm/src/component/soc/realtek/amebad/misc
...
+CFLAGS += -I$(TOPDIR)/board/rtl8720e/src/component/mbed/hal_ext/
+CFLAGS += -I$(TOPDIR)/board/rtl8720e/src/component/mbed/targets/hal/rtl8720e
+CFLAGS += -I$(TOPDIR)/board/rtl8720e/src/component/soc/amebalite/cmsis
+CFLAGS += -I$(TOPDIR)/board/rtl8720e/src/component/soc/amebalite/swlib
+CFLAGS += -I$(TOPDIR)/board/rtl8720e/src/component/soc/amebalite/app/monitor/include
+CFLAGS += -I$(TOPDIR)/board/rtl8720e/src/component/soc/amebalite/misc
+CFLAGS += -I$(TOPDIR)/board/rtl8720e/src/project/realtek_amebaLite_va0_example/inc/inc_km4
+CFLAGS += -I$(TOPDIR)/board/rtl8720e/src/component/soc/amebalite/fwlib/include
+CFLAGS += -I$(TOPDIR)/board/rtl8720e/src/component/ssl/mbedtls_ram_map/rom
+CFLAGS += -I$(TOPDIR)/board/rtl8720e/src/component/os/os_dep/include
```
Comment (sunghan-chang) [inline os/se/ameba/Make.defs:58]: Could you confirm they all are necessary indeed?

**Review by sunghan-chang — COMMENTED** [review]
You change the file permission of the km4_image3_all.bin to 755. Could you keep it as 644?

**Comment by sunghan-chang** [comment]
@taehwan-hwang @j11591 @hs36-kim Could you review SE changes?

File: build/configs/rtl8720e/hello/defconfig [hunk 3]
```diff
@@ -269,7 +269,20 @@ CONFIG_AUTOMOUNT_USERFS=y
...
+# CONFIG_SE_VIRTUAL is not set
+CONFIG_SE_AMEBA=y
+CONFIG_HW_RNG=y
+CONFIG_HW_DH_PARAM=y
+CONFIG_HW_ECDH_PARAM=y
+CONFIG_HW_RSA_VERIFICATION=y
+CONFIG_HW_ECDSA_VERIFICATION=y
+CONFIG_HW_RSA_ENC=y
+CONFIG_HW_SE_STORAGE=y
+CONFIG_SE_SECURE_CONTEXT_SIZE=4096
```
Comment (j11591) [inline build/configs/rtl8720e/hello/defconfig:285]: It is set as 6144 in TP1x. Is it right that this is different with TP1x?

File: os/se/ameba/Make.defs (hunk 2 above)
Comment (ziyik) [reply os/se/ameba/Make.defs:58]: Yes, these are necessary

File: build/configs/rtl8720e/hello/defconfig (hunk 3 above)
Comment (sunghan-chang) [reply build/configs/rtl8720e/hello/defconfig:285]: @ZhenBei-Sin As I remember, it was increased to 6144 to support one more additional algorithm. Please check it.

Comment (sunghan-chang) [reply build/configs/rtl8720e/hello/defconfig:285]: https://github.com/Samsung/TizenRT/commit/6f2f189738c127935b440c6814b39ce97b7f292f
This is for ED25519 support. @j11591 @sangwon03 @an4967 @taehwan-hwang Could you confirm ED25519 support is necessary?

Comment (ziyik) [reply build/configs/rtl8720e/hello/defconfig:285]: Yes, 6144 is better, as the default has been changed to 6144
We will change it to 6144

**Comment by sunghan-chang** [comment]
@ZhenBei-Sin @ziyik  There are conflicts. Could you check them?

File: os/se/ameba/rtl_security_api_wrapper_tz.c [hunk 4]
```diff
@@ -74,11 +74,14 @@ static int ns_flash_write(uint32_t address, uint32_t len, uint8_t *data)
 
 static void ns_setstatusbits(u32 NewState)
 {
+#ifdef CONFIG_AMEBAD_TRUSTZONE
 	device_mutex_lock(RT_DEV_LOCK_FLASH);
 	FLASH_Write_Lock();
 	FLASH_SetStatusBits(FLASH_STATUS_BITS, NewState);
 	FLASH_Write_Unlock();
 	device_mutex_unlock(RT_DEV_LOCK_FLASH);
+#else
```
Comment (kishore-sn) [inline os/se/ameba/rtl_security_api_wrapper_tz.c:83]: Since "#else" portion is empty, it can be removed.

Comment (taehwan-hwang) [inline os/se/ameba/rtl_security_api_wrapper_tz.c:83]:
```suggestion
```

**Comment by ZhenBei-Sin** [comment]
> @ZhenBei-Sin @ziyik There are conflicts. Could you check them?

Yes. We've already rebased it and resolved the conflict.

**Pull request merged**